src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.NonAndroid.cs

```diff
@@ -432,6 +500,19 @@ private static bool TryGetLocalTzFile([NotNullWhen(true)] out byte[]? rawData, [
             {
                 tzFilePath = tzVariable;
             }
+
+#if TARGET_WASI || TARGET_BROWSER
+            if (UseEmbeddedTzDatabase)
+            {
+                if(!TryLoadEmbeddedTzFile(tzFilePath, out rawData))
+                {
+                    return false;
+                }
+                id = tzVariable;
```

Comment:
The path is like `/usr/share/zoneinfo/Europe/Prague` and `id` is just `Europe/Prague` part.
In case of `UseEmbeddedTzDatabase` we could only support `id` in the `TZ` variable, not full path. 

I added condition for that, thanks for noticing!